### PR TITLE
adding histogram builder

### PR DESCRIPTION
**Pull request by JCOR11599**
Also removes unnecessary call to AnalyzeTable in information_schema

Companion PR: https://github.com/dolthub/dolt/pull/3365

### CODE REVIEW

File: sql/core.go [hunk 1]
```diff
@@ -421,6 +423,118 @@ type Histogram struct {
 // HistogramMap is a map from column name to associated histogram
 type HistogramMap map[string]*Histogram
 
+// HistogramMapBuilder will construct a HistogramMap given a PartitionIterator and Schema
+// TODO: have option for number of buckets (and logic to convert freqMap into those buckets)
+// TODO: could iterate over Partitions asynchronously (after exchange is rewritten)
+func HistogramMapBuilder(ctx *Context, t Table) (HistogramMap, error) {
```
Comment (max-hoffman) [inline sql/core.go:429]: this is written as a constructor, which we use `New...` to indicate. A builder is a class that exposes an interface for incrementally constructing the object, which when finished is returned with `builder.Build`. If you write it as a constructor, i would recommend taking `sql.RowIter` as an argument, not a table.

File: sql/core.go [hunk 2]
```diff
@@ -421,6 +423,118 @@ type Histogram struct {
 // HistogramMap is a map from column name to associated histogram
 type HistogramMap map[string]*Histogram
 
+// HistogramMapBuilder will construct a HistogramMap given a Table
```
Comment (zachmu) [inline sql/core.go:426]: Fix comment

File: sql/core.go [hunk 3]
```diff
@@ -421,6 +423,118 @@ type Histogram struct {
...
+// TODO: have option for number of buckets (and logic to convert freqMap into those buckets)
+// TODO: could iterate over Partitions asynchronously (after exchange is rewritten)
+func NewHistogramMapFromTable(ctx *Context, t Table) (HistogramMap, error) {
```
Comment (zachmu) [inline sql/core.go:429]: This method doesn't belong in core.go

Put this in a new file, statistics.go

**Review by zachmu — APPROVED** [review]
LGTM

**Pull request merged**